picard/ui/theme.py

```diff
@@ -96,11 +96,34 @@ def _missing_(cls, value):
         return cls.DEFAULT
 
 
-AVAILABLE_UI_THEMES = [UiTheme.DEFAULT]
-if IS_WIN or IS_MACOS:
-    AVAILABLE_UI_THEMES.extend([UiTheme.LIGHT, UiTheme.DARK])
-elif not IS_HAIKU:
-    AVAILABLE_UI_THEMES.extend([UiTheme.SYSTEM])
+def get_style_hints() -> QtGui.QStyleHints | None:
+    """Get style hints from QGuiApplication, returning None if unavailable."""
+    return QtGui.QGuiApplication.styleHints()
+
+
+def _style_hints_available() -> bool:
+    """Check if style hints are available on the current system."""
+    return get_style_hints() is not None
+
+
+# Theme availability based on platform capabilities
+if IS_HAIKU:
+    # Haiku doesn't support themes - UI is hidden anyway, but keep empty for consistency
+    # Platform Detection: `IS_HAIKU` is detected via sys.platform == 'haiku1'
```

Comment:
I think this line comment is not needed here. It is the same for all our `IS_{someos}` constants, which are there to avoid all the specific `sys.platform` tests.